packages/react-next/src/components/SwatchColorPicker/SwatchColorPicker.base.tsx

```diff
@@ -26,226 +24,93 @@ const getClassNames = classNamesFunction<ISwatchColorPickerStyleProps, ISwatchCo
 
 const COMPONENT_NAME = 'SwatchColorPicker';
 
-export class SwatchColorPickerBase extends React.Component<ISwatchColorPickerProps, ISwatchColorPickerState> {
-  public static defaultProps = {
-    cellShape: 'circle',
-    disabled: false,
-    shouldFocusCircularNavigate: true,
-    cellMargin: 10,
-  } as ISwatchColorPickerProps;
-
-  private _id: string;
-  private _cellFocused: boolean;
-
-  private navigationIdleTimeoutId: number | undefined;
-  private isNavigationIdle: boolean;
-  private readonly navigationIdleDelay: number = 250 /* ms */;
-  private async: Async;
+export const SwatchColorPickerBase = (props: ISwatchColorPickerProps) => {
+  const [selectedIndex, setSelectedIndex] = React.useState();
+  const id = props.id || getId('swatchColorPicker');
+  let cellFocused: boolean;
+  let currentSelectedIndex: number | undefined = undefined;
```

Comment:
Move the declaration of this to the spot where it's used (having it at the top is a bit misleading since it's essentially just a temp variable used in one spot)